drivers/hmis_external_apis/app/models/hmis_external_apis/external_forms/form_submission.rb

```diff
@@ -33,29 +33,35 @@ def self.from_raw_data(raw_data, object_key:, last_modified:, form_definition:)
         definition_id: form_definition.id,
         raw_data: raw_data,
       }
-      submission.save!
-      submission.process_custom_data_elements!(form_definition: form_definition)
+      FormSubmission.transaction do
```

Comment:
Thanks for adding some transactional safety! However it looks like we already wrap this in a tx where it's called from `ConsumeExternalFormSubmissionsJob` so we probably don't need both. Generally I think it's better to let the caller handle the tx if possible